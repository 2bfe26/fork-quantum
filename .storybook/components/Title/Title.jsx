import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import ColorPalette from '../../../components/Colors';

const StyledTitle = styled.h2`
  border-bottom: 1px solid ${ColorPalette.NEUTRAL.GRAY.WHITETHREE};
  padding-bottom: 8px;
  font-weight: normal;
  margin: 10px 0;
`;

const Title = ({ children, ...rest }) => (<StyledTitle {...rest}>{children}</StyledTitle>);

Title.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Title;
