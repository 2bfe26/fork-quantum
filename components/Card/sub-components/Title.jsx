import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Heading = styled.h2`
  font-size: ${props => (props.small ? '20px' : '24px')};
  font-weight: 600;
  margin: 0;
`;

const Title = ({ ...props }) => <Heading {...props} />;

Title.displayName = 'Card.Title';

Title.propTypes = {
  /** default `font-size` is `24px`, with `small` prop defined the `font-size` is changed to `20px`. */
  small: PropTypes.bool,
};

Title.defaultProps = {
  small: false,
};

export default Title;
