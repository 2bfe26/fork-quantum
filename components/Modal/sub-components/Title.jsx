import styled from 'styled-components';
import Card from '../../Card';

const Title = styled(Card.Title)`
  font-weight: 700;
  line-height: 1.25;
`;

export default Title;
