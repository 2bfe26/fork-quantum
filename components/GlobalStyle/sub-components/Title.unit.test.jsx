import React from 'react';
import renderer from 'react-test-renderer';
import Title from './Title';

describe('<Title />', () => {
  it('Should match the snapshot', () => {
    expect(renderer.create(<Title />).toJSON()).toMatchSnapshot();
  });
});
