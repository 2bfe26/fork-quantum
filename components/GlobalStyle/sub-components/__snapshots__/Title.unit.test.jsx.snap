// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> Should match the snapshot 1`] = `
<DocumentFragment>
  <h1
    class="Title-sc-1vnr32k-0"
  />
</DocumentFragment>
`;
